packages/bpk-component-page-indicator/src/BpkPageIndicator.js

```diff
@@ -0,0 +1,143 @@
+
+import NavButton, { DIRECTIONS } from './NavButton';
+import STYLES from './BpkPageIndicator.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+const DISPLAYED_TOTAL = 5;
+const START_SCROLL_INDEX = Math.floor(DISPLAYED_TOTAL / 2);
+
+export type Props = {
+  className: ?string,
+  showNav: ?boolean,
+  currentIndex: number,
+  totalBullets: number,
+  onClick: ?() => void,
+  bulletLabel: string,
+  prevNavLabel: string,
+  nextNavLabel: string,
+  dark: ?boolean,
+};
+
+const BpkPageIndicator = ({
+  bulletLabel,
+  className,
+  currentIndex,
+  dark,
+  nextNavLabel,
+  onClick,
+  prevNavLabel,
+  showNav,
+  totalBullets,
+}: Props) => (
+  // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
+  <div className={className}>
+    <div
+      className={getClassName(
+        'bpk-page-indicator',
+        showNav && 'bpk-page-indicator__showNav',
+      )}
+    >
+      {showNav && (
+        <NavButton
+          currentIndex={currentIndex}
+          onClick={onClick}
+          disabled={currentIndex === 0}
+          prev
+          ariaLabel={prevNavLabel}
+        />
+      )}
+      <div className={getClassName('bpk-page-indicator__container')}>
+        <div
+          className={getClassName('bpk-page-indicator__bullets-container')}
+          style={
+            currentIndex > START_SCROLL_INDEX
+              ? {
+                  '--scroll-index':
+                    totalBullets > DISPLAYED_TOTAL
+                      ? Math.min(
+                          currentIndex - START_SCROLL_INDEX,
+                          totalBullets - DISPLAYED_TOTAL,
+                        )
+                      : 0,
+                }
+              : undefined
+          }
+        >
+          {[...Array(totalBullets)].map((val, index) => (
+            <button
+              type="button"
+              onClick={(e) => {
+                onClick(e, index, DIRECTIONS.BULLETS);
+              }}
+              className={getClassName(
+                'bpk-page-indicator__bullet',
+                dark && 'bpk-page-indicator__dark',
+                index === currentIndex && 'bpk-page-indicator__bullet--active',
+                index === currentIndex &&
+                  dark &&
+                  'bpk-page-indicator__dark--active',
+              )}
+              aria-label={`${bulletLabel} ${index + 1}`}
+              aria-pressed={currentIndex === index ? 'true' : 'false'}
+              // eslint-disable-next-line react/no-array-index-key
+              key={`bullet-${index}`}
+            />
+          ))}
+        </div>
+      </div>
+      {showNav && (
+        <NavButton
+          currentIndex={currentIndex}
+          onClick={onClick}
+          disabled={currentIndex === totalBullets - 1}
+          ariaLabel={nextNavLabel}
+        />
+      )}
+    </div>
+  </div>
+);
+
+BpkPageIndicator.propTypes = {
+  bulletLabel: PropTypes.string.isRequired,
```

Comment:
Would you be able to use indicatorLabel instead of bulletLabel just so it clears any potential confusion on what that is and fits with the component?

Reply:
Have updated and changed from `bullet` into `indicator`
`bulletLabel` -> `indicatorLabel`
`totalBullets` -> `totalIndicators`
`__bullet` -> `__indicator`
Thank you very much!